lib/rom/rails/model/validator/uniqueness_validator.rb

```diff
@@ -24,13 +24,15 @@ def initialize(options)
           super
           @klass = options.fetch(:class)
           @message = options.fetch(:message) { :taken }
+          @scope_key = options[:scope]
         end
 
         # Hook called by ActiveModel internally
         #
         # @api private
         def validate_each(validator, name, value)
-          validator.errors.add(name, message) unless unique?(name, value)
+          scope = {@scope_key => validator.to_model[@scope_key]} if @scope_key
+          validator.errors.add(name, message) unless unique?(name, value, scope)
```

Comment:
`:scope` can be an array of attribute names, so we should support that. Probably something like that would do the trick:

``` ruby
names = Array(@scope_key) # probably best to set in the constructor
scope = Hash[names.zip(names.map {|name| self[name] })].merge(name => value)

validatos.errors.add(name, message) unless relation.unique?(scope)
```